documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -261,4 +266,59 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
 
 **Examples**: N/A
 
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S4002" />S4002	LocationMustHaveXmsMutability
+**Output Message**: Property 'location' must have '\"x-ms-mutability\":[\"read\", \"create\"]' extension defined. Resource Model: '{0}'
+
+**Description**: A tracked resource's `location` property must have the `x-ms-mutability` properties set as `read`, `create`.
+
+**Why the rule is important**: Location is a property that is set once and non-updatable for a tracked resource. Hence, per ARM guidelines the only operations allowed are `read` and `create`.
+
+**How to fix the violation**: Ensure that the `location` property in the tracked resource's hierarchy is correctly set to `read` and `create`.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S2008" />S2008	PostOperationIdContainsUrlVerb
+**Output Message**: OperationId must contain the verb: '{0}' in:'{1}'
+
+**Description**: A POST operation's operationID must contain the verb indicated at the end of the corresponding url.
+
+**Why the rule is important**: The url indicates the action performed by it, the corresponding POST operationId must therefore contain this verb for semantic consistency.
+
+**How to fix the violation**: Ensure that the operationId for POST operation contains the verb indicated at the end of the url.
```

Comment:
Can you include impact on generated code? In this case you can say the name of the generated method for that operation will pick up this name.